src/algorithms/calorimetry/CalorimeterHitReco.cc

```diff
@@ -35,6 +35,16 @@ void CalorimeterHitReco::AlgorithmInit(std::shared_ptr<spdlog::logger>& logger)
         return;
     }
 
+    // First, try and get the IDDescriptor. This will throw an exception if it fails.
+    try{
+        auto id_spec = m_geoSvc->detector()->readout(m_readout).idSpec();
```

Comment:
Doesn't this limit scope to the try block? Or at least complain about unused id_spec? And shadowing? How strict are you running, `-Werror`?